# HG changeset patch
# User sh@sh-laptop
# Date 1208369999 14400
#      Wed Apr 16 14:19:59 2008 -0400
# Node ID 83ea845cda36586912c35a704fc2727ca988d0dd
# Parent  050f695c3a7807fa9b2430675aad26e983fc958a
Display a (hopefully) informative error message if gnuplot isn't found

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-16  Soren Hauberg  <hauberg@gmail.com>
+
+	* plot/__gnuplot_version__.m: Display error if gnuplot is not found.
+
 2008-04-15  John W. Eaton  <jwe@octave.org>
 
 	* plot/view.m: Get values from current axes if nargin == 0.
 
 2008-04-15  David Bateman  <dbateman@free.fr>
 
 	* plot/__patch__.m: Fix for NaN values in faces of patches.
 	
diff --git a/scripts/plot/__gnuplot_version__.m b/scripts/plot/__gnuplot_version__.m
--- a/scripts/plot/__gnuplot_version__.m
+++ b/scripts/plot/__gnuplot_version__.m
@@ -23,16 +23,19 @@
 ## versions of gnuplot during the same session.
 
 function version = __gnuplot_version__ ()
 
   persistent __version__ = "";
 
   if (isempty (__version__))
     [status, output] = system (sprintf ("%s --version", gnuplot_binary ()));
+    if (status != 0)
+      error ("you must have gnuplot installed to display graphics; if you have gnuplot installed in a non-standard location, see the 'gnuplot_binary' function");
+    endif
     pattern = "^[^\\s]*\\s*([0-9]+\\.[0-9]+)\\s*[^\\s]*\\s*([^\\s]*)";
     [d1, d2, d3, d4, matches] = regexp (output, pattern);
     if (iscell (matches) && numel (matches) > 0 && iscellstr (matches{1}))
       __version__ = matches{1}{1};
     endif
   endif
 
   version = __version__;
